source/extensions/extensions_build_config.bzl

```diff
@@ -329,6 +329,12 @@ EXTENSIONS = {
 
     # apple DNS resolver extension is only needed in MacOS build plus one want to use apple library for DNS resolving.
     "envoy.network.dns_resolver.apple":                "//source/extensions/network/dns_resolver/apple:config",
+
+    #
+    # Custom matchers
+    #
+
+    "trie-matcher":                                    "//source/extensions/common/matcher:trie_matcher_lib",
```

Comment:
Those extensions should still follow the naming convention? like `envoy.matching.trie_matcher`?

Reply:
They should, but rename could potentially break someone who looks up extension by name.

Reply:
CC @snowp. I don't think there is a risk of collision since extensions are always qualified by their category. But we should rename them for consistency probably.

Reply:
Is lookup by name a thing anymore now that we're using typed_config? Isn't lookup first done by type if specified?

If we want to be really careful can we rename with runtime flag to revert to the old name per deprecation policies?

Reply:
Yeah, we are discouraging lookup by name. Speaking of which, we should really deprecate that lookup in more places instead of renaming extensions, and then name becomes purely internal.

Reply:
In any case, the name here does't affect lookup, so we should use a canonical name here and `extensions_metadata.yml`

Reply:
Wouldn't be mis-leading to document one extension name when by-name look up uses a different one? For this particular one, I agree, let's just change the name since it's new.

Reply:
Done.